[VAULT-20073] Docs: update docs on Sentinel RGP group policy application (#23292)
hashicorp/vault: website/content/api-docs/system/config-group-policy-application.mdx

@include 'alerts/restricted-root.mdx'

<Warning>

>   Starting with Vault versions `1.15.0`, `1.14.4`, and `1.13.8` the group policy application mode
>   setting applies only to ACL policies. Vault Sentinel Role Governing Policies (RGPs) are not affected
>   by group policy application mode.

Reviewer: ```suggestion
  The group policy application mode only applies to ACL policies and no longer
  affects Vault sentinel role governing policies (RGPs) for the following Vault
  versions:

  - `1.15.0+`
  - `1.14.4+`
  - `1.13.8+`

```

Author: Thanks, Sarah!
